service-proxy.js

```diff
@@ -626,6 +636,61 @@ function isExitFor(serviceName) {
     return chan.serviceProxyMode === 'exit';
 };
 
+ServiceDispatchHandler.prototype.emitPeriodicStats =
+function emitPeriodicStats() {
+    var self = this;
+    self.periodicStatsHandle = null;
+
+    var serviceNames = Object.keys(self.channel.subChannels);
+    for (var index = 0; index < serviceNames.length; index++) {
+        var serviceName = serviceNames[index];
+        var serviceChannel = self.channels.subChannels[serviceName];
+        self.emitPeriodicServiceStats(serviceChannel, serviceName);
+    }
+
+    self.requestPeriodicStats();
+};
+
+ServiceDispatchHandler.prototype.emitPeriodicServiceStats =
+function emitPeriodicServiceStats(serviceChannel, serviceName) {
+    var self = this;
+
+    var incoming = 0;
+    var outgoing = 0;
+    var anyway = 0;
+
+    var prefix = 'services.' + clean(serviceName, 'no-service') + '.';
+
+    var hostPorts = serviceChannel.peers.keys();
+    for (var pIndex = 0; pIndex < hostPorts.length; pIndex++) {
+        var hostPort = hostPorts[pIndex];
+        var peer = serviceChannel.peers.get(hostPort);
+        anyway += peer.connections.length;
+        for (var cIndex = 0; cIndex < peer.connections.length; cIndex++) {
+            var connection = peer.connections[cIndex];
+            if (connection.direction === 'in') {
+                incoming++;
+            } else if (connection.direction === 'out') {
+                outgoing++;
+            }
+        }
+    }
+
+    self.statsd.gauge(prefix + 'peers', hostPorts.length);
+    self.statsd.gauge(prefix + 'connections.in', incoming);
+    self.statsd.gauge(prefix + 'connections.out', outgoing);
+    self.statsd.gauge(prefix + 'connections.any', anyway);
+};
+
+ServiceDispatchHandler.prototype.requestPeriodicStats =
+function requestPeriodicStats() {
+    var self = this;
+    if (self.periodicStatsHandle) {
+        return;
+    }
+    self.periodicStatsHandle = self.channel.timers.setTimeout(self.boundEmitPeriodicStats, self.statsPeriod);
```

Comment:
wrap this in a `if (!self.channel.destroyed)` so that we don't hang process restarts